ocaml/xapi/xapi_vdi.ml

```diff
@@ -517,7 +519,8 @@ let destroy ~__context ~self =
       List.iter (fun vbd ->
           Helpers.log_exn_continue (Printf.sprintf "destroying VBD: %s" (Ref.string_of vbd))
             (fun vbd -> Db.VBD.destroy ~__context ~self:vbd) vbd) vbds;
-      (* Db.VDI.destroy ~__context ~self *)
+      (* If VDI destroyed is suspend VDI of VM then set the suspend_VDI field as null ref *)
+      List.iter (fun self -> Db.VM.set_suspend_VDI ~__context ~self ~value:Ref.null) suspend_vdi_vms;
```

Comment:
Oh, that's frustrating.
Or could we make `VDI.suspend_VDI_of` be `VM set ref` and do it that way? Not if the automatic updating works in one direction only... and anyway I don't like the taste of having a set (a list) where we want a zero-or-one.
Sigh. Perhaps we should keep this as it is, then (except for moving the definition of `suspend_vdi_vms` to just before we use it).